### PR TITLE
WELD-2264 Implement BeforeBeanDiscovery.configureQualifier() and configureInterceptorBinding() [C...

### PR DESCRIPTION
**Pull request by antoinesd**
WELD-2264 Implement BeforeBeanDiscovery.configureQualifier() and configureInterceptorBinding() [C...

### CODE REVIEW

**Comment by antoinesd** [comment]
TCK need an update to assert this new feature see https://issues.jboss.org/browse/CDITCK-554

**Comment by manovotn** [comment]
test this please

**Comment by mkouba** [comment]
test this please

File: impl/src/main/java/org/jboss/weld/bootstrap/events/BeforeBeanDiscoveryImpl.java [hunk 1]
```diff
@@ -143,14 +152,33 @@ public void addInterceptorBinding(AnnotatedType<? extends Annotation> bindingTyp
 
     @Override
     public <T extends Annotation> AnnotatedTypeConfigurator<T> configureQualifier(Class<T> qualifier) {
-        // TODO WELD-2264
-        return null;
+        checkWithinObserverNotification();
+        AnnotatedTypeConfiguratorImpl<T> configurator = new AnnotatedTypeConfiguratorImpl<>(getBeanManager().createAnnotatedType(qualifier));
+        additionalQualifiers.add(configurator);
```
Comment (mkouba) [inline impl/src/main/java/org/jboss/weld/bootstrap/events/BeforeBeanDiscoveryImpl.java]: Pls log the extension actions - see for example `BeforeBeanDiscoveryImpl.addQualifier(AnnotatedType<? extends Annotation>)`:

     BootstrapLogger.LOG.addQualifierCalled(getReceiver(), qualifier);

**Comment by antoinesd** [comment]
Yes @mkouba I didn't remove the exclusion.

File: impl/src/main/java/org/jboss/weld/bootstrap/events/BeforeBeanDiscoveryImpl.java (hunk 1 above)
Comment (mkouba) [inline impl/src/main/java/org/jboss/weld/bootstrap/events/BeforeBeanDiscoveryImpl.java]: Yes, please add all the missing logs to this PR.

Comment (antoinesd) [inline impl/src/main/java/org/jboss/weld/bootstrap/events/BeforeBeanDiscoveryImpl.java]: done

**Comment by mkouba** [comment]
retest this please

**Comment by mkouba** [comment]
Merged, thanks!

**Pull request closed without merging**